verification/curator-service/api/src/controllers/auth.ts

```diff
@@ -14,27 +20,49 @@ export const mustBeAuthenticated = (
     next: NextFunction,
 ): void => {
     if (req.isAuthenticated()) {
-        next();
-        return;
+        return next();
     }
     res.sendStatus(403);
 };
 
-// mustHaveAnyRole is an express middleware that checks that the currently logged-in user has any of the given roles.
+const userHasRequiredRole = (
+    user: UserDocument,
+    requiredRoles: Set<string>,
+): boolean => {
+    return user.roles?.filter((r: string) => requiredRoles.has(r)).length > 0;
```

Comment:
I wrote that but now I look again at it and I think we can use some() instead of building the whole filtered array and checking the length

Reply:
SG, done.